framework/update/internal/appupdateservice.cpp

```
@@ -216,15 +226,41 @@ const RetVal<ReleaseInfo>& AppUpdateService::lastCheckResult() const
 
 RetVal<Progress> AppUpdateService::downloadRelease()
 {
+    if (m_downloadInProgress) {
+        return RetVal<Progress>::make_ok(m_updateProgress);
+    }
+
     if (!m_networkManager) {
         m_networkManager = networkManagerCreator()->makeNetworkManager();
     }
 
     const ReleaseInfo info = m_lastCheckResult.val;
```

Clang (14.0.6)

[warning] variable 'info' is not initialized

(cppcoreguidelines-init-variables)